src/components/tables/Store.ts

```diff
@@ -115,50 +140,81 @@ export const getInitialState = (
-                    const statsData = {};
-                    data.forEach((datum) => {
-                        const { catalog_name } = datum;
-                        const currentStat = statsData[catalog_name];
-
-                        if (currentStat) {
-                            Object.entries(currentStat).forEach(
-                                ([key, value]) => {
-                                    if (typeof value === 'number') {
-                                        currentStat[key] =
-                                            currentStat[key] || 0;
-                                        currentStat[key] += datum[key];
+                } else if (data) {
+                    if (data.length > 0) {
+                        const statsData = {};
+                        data.forEach((datum) => {
+                            const { catalog_name } = datum;
+                            const currentStat = statsData[catalog_name];
+
+                            if (currentStat) {
+                                Object.entries(currentStat).forEach(
+                                    ([key, value]) => {
+                                        if (typeof value === 'number') {
+                                            currentStat[key] =
+                                                currentStat[key] || 0;
+                                            currentStat[key] += datum[key];
+                                        }
                                     }
-                                }
-                            );
-                        } else {
-                            statsData[catalog_name] = datum;
-                        }
-                        return datum;
-                    });
+                                );
+                            } else {
+                                statsData[catalog_name] = datum;
+                            }
+                            return datum;
+                        });
 
-                    console.log('setting stats', statsData);
-                    set(
-                        produce((state) => {
-                            state.stats = statsData;
-                        }),
-                        false,
-                        'Table Store Stats Hydration Success'
-                    );
+                        set(
+                            produce((state) => {
+                                state.stats = statsData;
+                            }),
+                            false,
+                            'Table Store Stats Hydration Success'
+                        );
+                    } else {
+                        set(
+                            produce((state) => {
+                                state.stats = [];
+                            }),
+                            false,
+                            'Table Store Stats Hydration Success : Empty'
+                        );
```

Comment:
Making sure an empty response is handled and put into the `stats` array so we can check for this.